### ISSUE TITLE
Create sparkplug mapper

### ISSUE THREAD

**Issue by toewsar** [thin-edge/thin-edge.io#2263]
**Is your feature request related to a problem? Please describe.**
When connecting to a custom broker it always a problem to find a way how topic and measurements are defined.
Unfortunatly azure iot hub and cumulocity have a very constrained messaging format which doesn't fit to any custom broker.
With sparkplug B there is a specification which is widely spreaded and used in industrial environments.
Developing a mapper which maps the data to a sparkplug comaptible format will enable thin-edge.io for a lot more applications.

**Describe the solution you'd like**
I'd recommend to develop a mapper (+ bridge config) which can connect to any sparkplug aware broker like hiveMQ.

**Describe alternatives you've considered**
It could be developed as a "different component" without integration into the RUST codebase, but I think this could be a valuable extension.

**Additional context**
You'll find more informations on the [sparkplug website](https://sparkplug.eclipse.org/), or on the [HiveMQ website](https://www.hivemq.com/mqtt-sparkplug-essentials/).

**Comment by reubenmiller** [comment]
@toewsar yes that's definitely an interesting idea. The new design of tedge MQTT api definitely goes in the direction of sparkplug B spec, so the mapping might not be too difficult, however this is probably something that the core team won't be able to start exploring until after the 1.x release.

Though that being said, we're more than happy for contributors to create a PoC and discuss it in more detail.

**Comment by reubenmiller** [comment]
Closing as Sparkplug support is part of the Roadmap. There is still interest into implement something, but there hasn't been much customer demand for it at this point in time.
